packages/neovim/build.sh

```diff
@@ -26,7 +26,8 @@ TERMUX_PKG_CONFFILES="share/nvim/sysinit.vim"
 termux_pkg_auto_update() {
 	# Get the latest release tag:
 	local tag
-	tag="$(termux_github_api_get_tag "${TERMUX_PKG_SRCURL}")"
+	tag="$(termux_github_api_get_tag "${TERMUX_PKG_SRCURL}" \
+		latest-regex "${TERMUX_PKG_UPDATE_VERSION_REGEXP}")"
```

Comment:
`^` and `$` around regex are missing.

Reply:
I had intentionally not done that - looks like the original behavior was a substring match, and I preserved it. It's not obvious to me that we'd want to reject `v0.9.2-webpfix`, or that you'd want a human to have to come back here if they switched to `V0.9.2` or `stable-0.9.2` or something.

Reply:
None of their version tags have a suffix and versions with suffix that do not exactly match suffix would fail on next line anyways since it has start and end anchors.

https://github.com/termux/termux-packages/blob/c25eb1c81d3791f05a76513208944deca49d4f83/packages/neovim/build.sh#L31

https://github.com/neovim/neovim/tags

Reply:
Oops, lost a piece of the change. Half the point here is for the `build.sh` to not have to worry about this and the tag getter will always return a matching tag or error out, so that goes away in any case. Thanks for noticing.

I was making a point that being overly-specific doesn't actually seem good, but I think I talked myself out of it. Made one more tweak to allow the package to specify anchors and then added them to its regex.

Reply:
Welcome. The `TERMUX_PKG_UPDATE_VERSION_REGEXP` is later used again, but without anchors, so your change should not break that, not sure why all packages are not directly adding it in variable. If versioning scheme does change for a package, and doesn't exactly match anymore, we should be notified of it, so makes sense to add anchors. Your pull requests fine now, if no one objects in a while, then can merge it.

https://github.com/termux/termux-packages/blob/9ea804f8f32ed1c5975982e552e546634f2cca3d/scripts/updates/utils/termux_pkg_upgrade_version.sh#L23

That `sed 's/^v//'` and later removal is potential trouble as more and more packages get added though.

Reply:
Note that this function already unconditionally strips leading `^v` from any tag it chooses before returning a version. The purpose of the sed was to avoid having the regex include the `v` (which IIRC screwed up the actual do-the-auto-updater, since the returned version wouldn't match the regex) or having this function insert the v into the given regex, after the anchor you asked for (which sucks).

If this lgty, I suggest merging it - 0.9.0 is sorta a little materially broken (`nvim-treesitter` is now demanding `0.9.1`, apparently). Obviously we could bump it manually, but it would be nice to see this work e2e and fix that in one go.

Reply:
>Note that this function already unconditionally strips leading ^v 

That's what I meant by "and later removal". Leaving it as is for now, if it becomes a problem, will just fetch the original author ;)

Please following commit guidelines in future and create separate commits for script and package changes.

https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#commit-guidelines

I triggered an auto update and `0.9.2` has been picked up,  and is available at default repo, and mirrors when they sync.

https://github.com/termux/termux-packages/actions/runs/6365118938

https://github.com/termux/termux-packages/actions/runs/6365175235

https://packages.termux.dev/apt/termux-main/pool/main/n/neovim/

> If this lgty, I suggest merging it 

Not everyone can merge this pull, for example grimler wouldn't touch anything related to vim, he's from the church of emacs ;)

Reply:
> Please following commit guidelines in future
Ack. Sorry, was thinking in unit-of-rollback terms. :)

> I triggered an auto update and 0.9.2 has been picked up
Appreciate it. Now I don't have to live in suspense. :)